internal/stacks/stackruntime/internal/stackeval/component_instance.go

```diff
@@ -1591,3 +1592,7 @@ func (c *ComponentInstance) tracingName() string {
 func (c *ComponentInstance) reportNamedPromises(cb func(id promising.PromiseID, name string)) {
 	cb(c.moduleTreePlan.PromiseID(), c.Addr().String()+" plan")
 }
+
+func (c *ComponentInstance) PlanTimestamp() time.Time {
```

Comment:
nit: can you move this next to the `ResolveExpressionReference` function, and add the same doc comments around it implementing ExpressionScope.